### ISSUE TITLE
Clean up design detail view

### ISSUE THREAD

**Issue by Woseseltops** [Signbank/Global-signbank#498]
Following the observation in #439 that there are a lot of things going on in the upper regions of the detail view, leading to a messy feel...

![](https://user-images.githubusercontent.com/1909317/40167162-9071c750-59c0-11e8-943b-013b905561e1.png)

... I forced myself to verbalize what was causing this feel, and this is what I came up with.

### Selected datasets
* In the rest of the design, a box with rounded corners is something you can click.
* Selected datasets is really something that goes for the entire site, like the selected language, but the fact that they are floating at the top of the page makes them (incorrectly) feel like this is something only relevant to this page.
* I feel like the combination of dark grey with a blue outline somehow doesn't fit the rest of the design, although I can't pinpoint why I feel that way - the colors themselves are used elsewhere.

### Button row: relations/public/delete
* The alignment is off
* This kind of lay out, with buttons stuck to each other, is often used for some kind of toggle for different 'states' or 'modes' you can be in. This fits perfectly with the views (the detail page can be in 'public mode', 'relations mode', etc)... but not with the Delete button.
* When selecting a mode, you would expect to see visualized somehow in which mode you are... but in our case the button disappears. For example, you can only see the the 'Public view' button when not in the public view. This is unlike the button row above it.

**Comment by Woseseltops** [comment]
An attempt to address these points. I'm not fully happy with it, but it at least shows in which direction I'm thinking. What's your view @ocrasborn ?

![detail_page_redesign](https://user-images.githubusercontent.com/1909317/45221943-53d29580-b2b3-11e8-8799-9cc3137a1a32.png)

**Comment by ocrasborn** [comment]
Yes, that's already much better! I somehow feel that the buttons for selecting a view are of a different kind of beast than the sign hits, even though you can click on both. Would making the views look like tabs, with a line below them, make sense?

**Comment by ocrasborn** [comment]
yeah! The lower two are much better. and maybe the addition of the red, although more prominent, is actually better in distinguishing the different roles of all these clickables.

**Comment by ocrasborn** [comment]
The result scroll bar I think we could still improve on. Less spacing above/below the text? Making them all pop out or sink in the page... there should be good design examples for this online i reckon

**Comment by Woseseltops** [comment]
Ha! Realized only now that Github uses the exact same tabs, even with the colored line on top. The design already looked familiar to me :) . 

> The result scroll bar I think we could still improve on. Less spacing above/below the text? Making them all pop out or sink in the page... there should be good design examples for this online i reckon

Not sure if I fully understand what you mean @ocrasborn . Something like this?

![image](https://user-images.githubusercontent.com/1909317/45546141-b7614380-b81c-11e8-9c8e-c40dfa3f91ca.png)

> I originally tried the datasets in that Radboud red and they looked awful. It's kind of a blood colour.

Yeah I tried a number of things there as well. How do you feel about the current proposal to visualize them, @susanodd ?

PS
Now we're talking design, I was thinking that maybe making the grey of the word 'signbank' in the header a bit more reddish would look prettier. Thoughts?

![image](https://user-images.githubusercontent.com/1909317/45544558-f04ae980-b817-11e8-8251-10bbe577e30a.png)

**Comment by ocrasborn** [comment]
yes, exactly! Maybe the non-selected ones a wee bit darker grey?

**Comment by susanodd** [comment]
Yes, I like that reddish signbank underneath the white Global.

**Comment by Woseseltops** [comment]
Like this?

![image](https://user-images.githubusercontent.com/1909317/45548897-28f1bf80-b826-11e8-9167-021a08241e36.png)

Anyway, this is something we can easily tweak until infinity also once it's been implemented... the rest of the design is okay with you guys?

**Comment by ocrasborn** [comment]
looks OK to me @Woseseltops , let's see how users respond to it

**Comment by susanodd** [comment]
Yes, it looks good!

**Comment by Woseseltops** [comment]
Todo:
* Switch around colors title bar
* Try datasets in the title bar

**Comment by Woseseltops** [comment]
Some experiment with the logo:

* Swapped colors

![image](https://user-images.githubusercontent.com/1909317/45868464-0c093f00-bd86-11e8-9cf6-e25e977fa3e3.png)

* Differences in font size and position

![image](https://user-images.githubusercontent.com/1909317/45868647-9c478400-bd86-11e8-9641-2b25946e8a76.png)

* Capital Signbank

![image](https://user-images.githubusercontent.com/1909317/45868875-25f75180-bd87-11e8-919e-bac99bd2741a.png)

**Comment by Woseseltops** [comment]
Some experiments with the datasets in the red bar:

* 3 datasets. I think I like it, because this way it's almost like an explanation of what 'global' refers to exactly.

![image](https://user-images.githubusercontent.com/1909317/45870079-60aeb900-bd8a-11e8-80dd-0cf60676c740.png)

* If you have a lot of datasets selected

![image](https://user-images.githubusercontent.com/1909317/45870164-8b990d00-bd8a-11e8-91f5-ba6022741827.png)

* If you want to make a difference between selected and available datasets

![image](https://user-images.githubusercontent.com/1909317/45870236-b84d2480-bd8a-11e8-839b-a090180fb1f0.png)

* Alternative, where selected is lit up instead of non-selected faded

![image](https://user-images.githubusercontent.com/1909317/45870299-d87ce380-bd8a-11e8-871f-23249b0ef640.png)

**Comment by ocrasborn** [comment]
Hi @woseseltops, let's go for the change in colours  of global Signbank like you propose, but keep the capitalisation and spacing as is.
As for the datasets, please implement the third of the four experiments: I think that has the best balance of subtlety and information.

**Comment by Woseseltops** [comment]
@ocrasborn , I've disabled some code showing search results on the public view (??)... you know if this is still used?

**Comment by ocrasborn** [comment]
Yes, it is, and will be: perhaps we need to configure this better or differently in the future, but most of the NGT dataset should become publicly accessible next year (checkmark with 'Publish in dictionary', so that we can share data yet don't need to handle hundreds or thousands of registration requests.

**Comment by Woseseltops** [comment]
Not sure if your answer 100% matches up with my question :) . 

What I mean is that when you the access the detail view of a gloss from the search results, you can see a list of other search results (you called this 'result scroll bar' above). I've encountered code that appears to be doing something similar for the public view of a gloss, where you can navigate other search results by clicking buttons... but I can't get it to actually show something. 

My question is if you recognize this functionality and/or know how it works. If not, it's probably some relic from the past that can be removed.

The code for completeness:

```
<div id="searchresults" class='navbar navbar-default' style="border:0;">
</div>
    <div id="signinfo" class='navbar navbar-default navbar-collapse'>
    
        <div class='btn-group'>
            {% if SIGN_NAVIGATION %}
        
                {% if navigation.prev %}
                {# Translators: Navigation  #}
                <a class='btn navbar-btn btn-default' href="{{PREFIX_URL}}/dictionary/gloss/{{navigation.prev.idgloss}}.html?lastmatch={{lastmatch}}">&laquo; <span class='hidden-xs'>{% blocktrans %}Previous Sign{% endblocktrans %}</span></a>
           
                {% endif %}
                {# Translators: Sign x of n(signs) #}
                <button class='btn navbar-btn'>{% blocktrans %}Sign {{glossposn}} of {{glosscount}} <span class='hidden-xs'>in the dictionary{% endblocktrans %}</span></button>
            
                {% if navigation.next %}
                {# Translators: Navigation #}
                <a class="btn navbar-btn btn-default"
                   href="{{PREFIX_URL}}/dictionary/gloss/{{navigation.next.idgloss}}.html?lastmatch={{lastmatch}}"><span class='hidden-xs'>{% blocktrans %}Next Sign{% endblocktrans %}</span> &raquo;</a>
                {% endif %}
             {% else %}
           
             {% endif %}
        </div>
   
   
            {% ifequal viewname "words" %}
            <div class='pull-right'>
            {# Translators: Navbar text #}
            <span class='navbar-text'>{% blocktrans %}Matches for the word <em>{{ translation|safe }}{% endblocktrans %}</em></span>
            <div class='btn-group'>
            {% for i in matches %}
                {% ifequal i n %}
                <button type='button' class='btn btn-primary navbar-btn'>{{i}}</button>
                {% else %}
                <a type='button' class='btn btn-default navbar-btn' href="{{translation|safe}}-{{i}}.html">{{i}}</a>
                {% endifequal %}
            {% endfor %}
            </div>
            {% else %}
                {% if lastmatch %}
                {# Translators: Navigation  #}
                <a class='btn btn-default navbar-btn' href="{{PREFIX_URL}}/dictionary/words/{{lastmatch}}.html">{% blocktrans %}Return to Matches{% endblocktrans %}</a>
                {% endif %}
            </div>
            {% endifequal %}
```

**Comment by Woseseltops** [comment]
@vanlummelhuizen dug into the edit history of Signbank for me today, and indeed showed that this is pre-NGT code that we probably don't use.

**Comment by Woseseltops** [comment]
Okay I think we're pretty close to the mockup. Happy @ocrasborn ?

**Comment by ocrasborn** [comment]
Yes, pretty close! Two things:
- The search results bar (Mac, Safari and Firefox) is partly hidden now. Removing the text (and its line) "Search results:" would make space again. We don't actually need that text, it's self-evident.
- The text Global Signbank is now very... I wouldn't use the word 'bold', but rather 'obese'. Could you find a way to make it more stylish again, like it was before?

**Comment by Woseseltops** [comment]
> The search results bar (Mac, Safari and Firefox) is partly hidden now. 

Not really sure what you mean here, for me it looks okay in all browsers. Can you make a screenshot (or ask me to come to your office so you can show)?

> The text Global Signbank is now very... I wouldn't use the word 'bold', but rather 'obese'. Could you find a way to make it more stylish again, like it was before?

I don't think that text changed (boldness is still 700), but perhaps now it has less bold things next to it the boldness grabs your attention. 

This is what it looks like with boldness 400:

![image](https://user-images.githubusercontent.com/1909317/48252130-2c5b8e80-e404-11e8-91b6-8c7a5c6f0788.png)

And boldness 600:

![image](https://user-images.githubusercontent.com/1909317/48252281-a0963200-e404-11e8-9b49-c86d56e2c8dc.png)

**Comment by ocrasborn** [comment]
Hi @Woseseltops : hm. Perhaps you're right. The version live now is good enough for me. Today :-)
The screenshot from Safari:
<img width="1027" alt="screenshot 2018-11-09 10 34 47" src="https://user-images.githubusercontent.com/10242207/48254918-8dd32b80-e40b-11e8-8fa9-98f401a310f8.png">

**Comment by Woseseltops** [comment]
> The screenshot from Safari:

Hmm yeah weird I cannot reproduce on Ubuntu. Perhaps together with Micha next week.

**Comment by susanodd** [comment]
The Search Results label ended up underneath the stuff above it in the most recent version.

**Comment by susanodd** [comment]
Oh, it's visible in the screenshot above. The sign bar is hidden by the Search Results label.

**Comment by susanodd** [comment]
The horizontal scroll bar under the search results was hidden.
I revised this so it shows again. If you do a search that yields many glosses (such as "ge" for NGT), then the scroll bar needs to be visible.
One problem that I don't know how to solve, if you Edit the Gloss, and then scroll down, the space around the search results is transparent and the Edit screen can be seen behind it, between the Search Results and the Navigation Bar. I don't know if that's a problem or not. I tried to get rid of it a long time ago, but that's how there ended up so much white space.
It would be nice if the display could adjust depending on whether there is a horizontal scroll bar.

![search-results-scroll-bar-visible](https://user-images.githubusercontent.com/24272544/48774328-34e87a80-ecca-11e8-99df-dddcdc015344.png)

![edit-showing-behind-search-results](https://user-images.githubusercontent.com/24272544/48774259-05397280-ecca-11e8-8e8d-969de6defe54.png)
